Library/Homebrew/cmd/cleanup.rb

```diff
@@ -129,6 +123,24 @@ def rm_DS_Store
     quiet_system "find", *args
   end
 
+  def prune_flag(default = nil)
+    # Construct a flag for use with `prune?` from argument '--prune=<days>'
+    # (with 'all' meaning to prune regardless of age) with fallback to default
+    # value or not pruning at all (in absence of a default).
+    prune = ARGV.value "prune"
+    if prune == "all"
+      "all"
+    elsif prune
+      Time.now - 60 * 60 * 24 * prune.to_i
+    elsif default
+      Time.now - 60 * 60 * 24 * default
+    end
+  end
+
+  def prune?(time, flag)
+    flag && ((flag == "all") || (time < flag))
```

Comment:
Use an `if` statement and have this function take a single argument, get the value from `ARGV` unless it's set and call it `cleanup`